File: 251115/키, 몸무게를 기준으로 정렬/sort-by-height-and-weight.py
class Student:
    def __init__(self, name, height, weight):
        self.name = name
        self.height = height
        self.weight = weight

n = int(input())
students = []

for _ in range(n):
    n, h, w = input().split()
    students.append(Student(n, int(h), int(w)))

students.sort(key=lambda x: (x.height, -x.weight))

for s in students:
    print(f'{s.name} {s.height} {s.weight}')

File: 260224/키, 몸무게를 기준으로 정렬/sort-by-height-and-weight.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split("\n");

const n = Number(input[0]);

class Person {
    constructor(name, height, weight) {
        this.name = name;
        this.height = height;
        this.weight = weight;
    }
}

const cmp = (a, b) => {
    if (a.height === b.height) {
        return b.weight - a.weight;
    }

    return a.height - b.height;
}

const personList = [];
for (let i = 1; i <= n; i++) {
    const [name, height, weight] = input[i].split(" ");
    personList[i - 1] = new Person(name, Number(height), Number(weight));
}

personList.sort(cmp);

for (let i = 0; i < n; i++) {
    console.log(`${personList[i].name} ${personList[i].height} ${personList[i].weight}`);
}